src/framework/theme/components/form-field/form-field.component.ts

```diff
@@ -0,0 +1,219 @@
+ * form-field-addon-medium-line-height:
+ * form-field-addon-medium-font-weight:
+ * form-field-addon-large-height:
+ * form-field-addon-large-width:
+ * form-field-addon-large-icon-size:
+ * form-field-addon-large-font-size:
+ * form-field-addon-large-line-height:
+ * form-field-addon-large-font-weight:
+ * form-field-addon-giant-height:
+ * form-field-addon-giant-width:
+ * form-field-addon-giant-icon-size:
+ * form-field-addon-giant-font-size:
+ * form-field-addon-giant-line-height:
+ * form-field-addon-giant-font-weight:
+ **/
+@Component({
+  selector: 'nb-form-field',
+  styleUrls: ['./form-field.component.scss'],
+  templateUrl: './form-field.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NbFormFieldComponent implements AfterContentChecked, AfterContentInit, OnDestroy {
+
+  protected readonly destroy$ = new Subject<void>();
+
+  protected formControlState$ = new BehaviorSubject<NbFormControlState>({
+    status: 'basic',
+    size: 'medium',
+    focused: false,
+    disabled: false,
+  });
+  prefixClasses$: Observable<string[]> = this.getAddonClassesObservable('prefix');
+  suffixClasses$: Observable<string[]> = this.getAddonClassesObservable('suffix');
+
+  @ContentChildren(NbPrefixDirective, { descendants: true }) prefix: QueryList<NbPrefixDirective>;
+  @ContentChildren(NbSuffixDirective, { descendants: true }) suffix: QueryList<NbSuffixDirective>;
+
+  @ContentChild(NbInputDirective, { static: false }) formControl: NbInputDirective;
+  @ContentChild(NbInputDirective, { static: false, read: ElementRef }) formControlElement: ElementRef<HTMLInputElement>;
+
+  constructor(
+    protected cd: ChangeDetectorRef,
+    protected renderer: Renderer2,
+  ) {
+  }
+
+  ngAfterContentChecked() {
+    if (!this.formControl) {
+      throwFormControlElementNotFound();
+    }
+
+    this.checkFormControlState();
+  }
+
+  ngAfterContentInit() {
+    this.subscribeToFormControlFocusChange();
+    this.subscribeToAddonChange();
+  }
+
+  ngOnDestroy() {
+    this.destroy$.next();
+  }
+
+  protected subscribeToFormControlFocusChange() {
+    const formControlEl = this.formControlElement.nativeElement;
+    fromEvent(formControlEl, 'focusin')
+      .pipe(
+        tap(() => this.updateFormControlState({ focused: true })),
+        switchMap(() => fromEvent(formControlEl, 'focusout')),
+        takeUntil(this.destroy$),
+      )
+      .subscribe(() => this.updateFormControlState({ focused: false }));
+  }
+
+  protected subscribeToAddonChange() {
+    merge(this.prefix.changes, this.suffix.changes)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => this.cd.markForCheck());
+  }
+
+  protected checkFormControlState() {
+    const formControlEl: HTMLElement = this.formControlElement.nativeElement;
+    const prevSize = this.formControlState$.value.size;
+    const size = this.formControl.fieldSize;
+
+    this.renderer.removeClass(formControlEl, this.getAddonSizeClass('prefix', prevSize));
```

Comment:
Are you sure it's the best solution to add classes to `formControlEl` instead of add classes to form-field and then write styles that will affect `input`?